File: EXPERIENCIA-DE-PRÁCTICA-1-ABSTRACCIÓN/EJERCICIO-1.1/claseAnimal.cpp
#include <iostream>
#include <string>

using namespace std;

// Definición de clase Animal
class Animal {
private:
    string nombre;
    int edad;
    string tipo;

public:
    // Métodos para establecer los valores de los atributos
    void setNombre(string _nombre) // Método para  el nombre del animal
    {
        nombre = _nombre;
    }

    void setEdad(int _edad) // Método para la edad del animal
    {
        edad = _edad;
    }

    void setTipo(string _tipo) // Método para el tipo del animal
    {
        tipo = _tipo;
    }

    // Métodos para obtener los valores de los atributos
    string getNombre() const // obtener el nombre del animal
    {
        return nombre;
    }

    int getEdad() const { // obtener la edad del animal
        return edad;
    }

    string getTipo() const { // obtener el tipo del animal
        return tipo;
    }

    // Métodos para realizar acciones con el animal
    void alimentar() { // alimentar al animal
        cout << nombre << " está siendo alimentado" << endl;
    }

    void hacerSonido() { // animal haga un sonido
        cout << nombre << " está haciendo sonido" << endl;
    }
};

int main() {
    Animal perro; // Instancia de un perro
    perro.setNombre("Platano"); // Establecer el nombre del perro
    perro.setEdad(5); // Establecer la edad del perro
    perro.setTipo("Perro"); // Establecer el tipo del perro

    Animal gato; // Instancia de un gato
    gato.setNombre("Tiburon"); // Establecer el nombre del gato
    gato.setEdad(3); // Establecer la edad del gato
    gato.setTipo("Gato"); // Establecer el tipo del gato

    // Mostrar información
    cout <<"\nMASCOTA 1:" <<endl; // Imprimir información sobre la mascota 1
    cout << "Nombre---> " << perro.getNombre() << "\nEdad ---> " << perro.getEdad() << "\nTipo---> " << perro.getTipo() << endl; // Imprimir nombre, edad y tipo del perro
    perro.alimentar(); // Alimentar al perro
    perro.hacerSonido(); // Hacer que el perro haga un sonido

    cout <<"\nMASCOTA 2:" <<endl; // Imprimir información sobre la mascota 2
    cout << "Nombre---> " << gato.getNombre() << "\nEdad ---> " << gato.getEdad() << "\nTip

File: EXPERIENCIA-DE-PRÁCTICA-1-ABSTRACCIÓN/EJERCICIO-1.1/clase animal.java
// Clase Animal: Representa a un animal con atributos como nombre, edad y tipo, y métodos para interactuar con el animal. 

public class Animal { 

    // Atributos 

    private String nombre; 

    private int edad; 

    private String tipo; 

 

    // Constructor 

    public Animal(String nombre, int edad, String tipo) { 

        this.nombre = nombre; 

        this.edad = edad; 

        this.tipo = tipo; 

    } 

 

    // Métodos para acceder a los atributos 

    public String getNombre() { 

        return nombre; 

    } 

 

    public int getEdad() { 

        return edad; 

    } 

 

    public String getTipo() { 

        return tipo; 

    } 

 

    // Método para alimentar al animal 

    public void alimentarAnimal() { 

        System.out.println(nombre + " fue alimentado."); 

    } 

 

    // Método para hacer sonidos 

    public void hacerSonidoAnimal() { 

        switch (tipo) { 

            case "perro": 

                System.out.println("¡Guau!"); 

                break; 

            case "gato": 

                System.out.println("¡Miau!"); 

                break; 

            case "oveja": 

                System.out.println("¡Beee!"); 

                break; 

            default: 

                System.out.println("No está en la lista."); 

        } 

    } 

 

    // Método main para probar la clase 

    public static void main(String[] args) { 

        // Creación de diferentes objetos para representar diferentes animales 

        Animal perro = new Animal("Chimuelo", 2, "perro"); 

        Animal gato = new Animal("Tarzán", 3, "gato"); 

        Animal oveja = new Animal("Dolly", 1, "oveja"); 

 

        // Interacción 

        perro.alimentarAnimal(); 

        perro.hacerSonidoAnimal(); 

 

        gato.alimentarAnimal(); 

        gato.hacerSonidoAnimal(); 

 

        oveja.alimentarAnimal(); 

        oveja.hacerSonidoAnimal(); 

    } 

} 
